packages/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,625 @@
+import React, { Component, Fragment, ReactNode, RefObject } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { emotion } from '@leafygreen-ui/lib';
+
+const { css, cx } = emotion;
+
+const rootPopoverStyle = css`
+  transition: transform 150ms ease-in-out, opacity 150ms ease-in-out;
+  position: absolute;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+type RefPosition = {
+  top: number;
+  bottom: number;
+  left: number;
+  right: number;
+  height: number;
+  width: number;
+};
+
+type AbsolutePositionObject = {
+  top?: string | number;
+  bottom?: string | number;
+  left?: string | number;
+  right?: string | number;
+};
+
+const defaultRefPosition = {
+  top: 0,
+  bottom: 0,
+  left: 0,
+  right: 0,
+  height: 0,
+  width: 0,
+};
+
+type Alignment = 'top' | 'bottom' | 'left' | 'right';
+type Justify = 'start' | 'middle' | 'end';
+
+// We transform 'middle' into 'center-vertical' or 'center-horizontal' for internal use,
+// So both Justify and Justification are needed, where the same is not true for Alignment.
+type Justification =
+  | 'top'
+  | 'bottom'
+  | 'left'
+  | 'right'
+  | 'center-vertical'
+  | 'center-horizontal';
+
+type AbstractPosition = {
+  alignment?: Alignment;
+  justification?: Justification;
+};
+
+type Props = {
+  children?: ReactNode;
+  active?: boolean;
+  className?: string;
+  align?: Alignment;
```

Comment:
If it's in `defaultProps`, then it should be marked as required. Defaults are applied before the types are checked.

As far as TS is concerned, for a component `Comp` with `defaultProps = { foo: 'bar' }`, `<Comp />` is equivalent to `<Comp foo="bar" />`.